library/core/src/fmt/num.rs

```diff
@@ -312,8 +320,8 @@ macro_rules! impl_Display {
 
         #[cfg(feature = "optimize_for_size")]
         fn $gen_name(mut n: $u, is_nonnegative: bool, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-            // 2^128 is about 3*10^38, so 39 gives an extra byte of space
-            let mut buf = [MaybeUninit::<u8>::uninit(); 39];
+            const SIZE: usize = $u::MAX.ilog(10) as usize + 1;
+            let mut buf = [MaybeUninit::<u8>::uninit(); SIZE];
```

Comment:
What is the effect of this change on binary size?

Reply:
On smaller targets, the maximum size might be of `u32` and not `u64` so since we're there... But on the binary size, there should be no difference for this change.